File: ensate-rh-backend/src/main/java/ma/ensate/rh/ensate_rh_backend/dto/PersonnelDTO.java
package ma.ensate.rh.ensate_rh_backend.dto;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class PersonnelDTO {
    
    private Long id;
    
    @Valid
    @NotNull(message = "Citizen information is required")
    private CitoyenDTO citoyen;
    
    @NotBlank(message = "PPR is required")
    private String ppr;
    
    private String typeEmploye;
    
    private LocalDate dateRecrutementMinistere;
    
    private LocalDate dateRecrutementEnsa;
    
    private String gradeActuel;
    
    private Integer echelleActuelle;
    
    private Integer echelonActuel;
    
    private Integer soldeConges = 0;
    
    private Boolean estActif = true;
}

File: FrontEnd/app/src/main/java/com/ensa/gestionpersonnel/data/remote/dto/PersonnelDto.kt
package com.ensa.gestionpersonnel.data.remote.dto

data class PersonnelDto(
    val id: Long? = null,
    val ppr: String,
    val citoyen: CitoyenDto,
    val typeEmploye: String,
    val dateRecrutementMinistere: String,
    val dateRecrutementEnsa: String, // Corrected casing from ENSA to Ensa
    val gradeActuel: String,
    val echelleActuelle: Int,
    val echelonActuel: Int,
    val soldeConges: Int = 30,
    val estActif: Boolean = true
)
